src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.Tests/CollectionMarshallingFails.cs

```diff
@@ -169,60 +176,81 @@ public void MultidimensionalArray_CheckOuterArrayIsIndexTracked()
                 });
-                EnforceAllElementsCleanedUpBoolStruct.AssertAllHaveBeenCleaned();
+                BoolStructInMarshaller.Marshaller.AssertAllHaveBeenCleaned();
+                BoolStructOutMarshaller.Marshaller.AssertAllHaveBeenCleaned();
             }
             // Run without throwing - this is okay only because the native code doesn't actually use the array, it creates a whole new one
-            EnforceAllElementsCleanedUpBoolStruct.ThrowOnNthUnmarshalledElement(-1);
-            EnforceAllElementsCleanedUpBoolStruct.ExpectedCleanupNumber = 10;
+            BoolStructOutMarshaller.Marshaller.UnmarshallingFailsIndex = -1;
+            BoolStructInMarshaller.Marshaller.ExpectedFreedValues = nativeArr;
+            BoolStructOutMarshaller.Marshaller.ExpectedFreedValues = nativeNegated;
             NativeExportsNE.MarshallingFails.NegateBoolsOut(arr, arr.Length, out var boolsOut);
-            EnforceAllElementsCleanedUpBoolStruct.AssertAllHaveBeenCleaned();
+            BoolStructInMarshaller.Marshaller.AssertAllHaveBeenCleaned();
+            BoolStructOutMarshaller.Marshaller.AssertAllHaveBeenCleaned();
         }
 
         [Fact]
         public void MultiDimensionalOutArray_EnsureAllCleaned()
         {
             var arr = GetMultiDimensionalArray<BoolStruct>(10, 10);
             var widths = new int[10] { 10, 10, 10, 10, 10, 10, 10, 10, 10, 10 };
-            //foreach (var throwOn in new int[] { 0, 1, 45, 99 })
-            //{
-            //    EnforceAllElementsCleanedUpBoolStruct.ThrowOnNthUnmarshalledElement(throwOn);
-            //    EnforceAllElementsCleanedUpBoolStruct.ExpectedCleanupNumber = 100;
-            //    Assert.Throws<ArgumentException>(() =>
-            //    {
-            //        NativeExportsNE.MarshallingFails.NegateBoolsOut2D(arr, arr.Length, widths, out BoolStruct[][] boolsOut);
-            //    });
-            //    EnforceAllElementsCleanedUpBoolStruct.AssertAllHaveBeenCleaned();
-            //}
+            foreach (var throwOn in new int[] { 0, 1, 45, 99 })
+            {
+                // Set up unmarshalling asserts
+                BoolStructOutMarshaller.Marshaller.UnmarshallingFailsIndex = throwOn;
+                BoolStructOutMarshaller.Marshaller.ExpectedFreedValues = Enumerable.Range(0, 100).Select(_ => new BoolStructNative() { b1 = 1, b2 = 1, b3 = 1 }).ToArray();
+                // https://github.com/dotnet/runtime/issues/93423
+                //NegateBoolStructInMarshaller.Marshaller.ExpectedFreedValues = Enumerable.Range(0, 100).Select(_ => new BoolStructNative() { b1 = 0, b2 = 0, b3 = 0 }).ToArray();
+                Assert.Throws<ArgumentException>(() =>
+                {
+                    NativeExportsNE.MarshallingFails.NegateBoolsOut2D(arr, arr.Length, widths, out BoolStruct[][] boolsOut);
+                });
+                // https://github.com/dotnet/runtime/issues/93423
```

Comment:
When tests are disabled with an issue, remember to add the "disabled tests" so when it is worked on someone looks of tests.